twoliter/embedded/imghelper

```diff
@@ -0,0 +1,111 @@
+#!/usr/bin/env bash
+
+# shellcheck disable=SC2034 # variables are used through parameter substitution
+symlink_image() {
+  local ext what target symlink_prefix version_id output_dir
+  ext="$1"
+  what="${2:?}"
+  symlink_prefix="${3:?}"
+  version_id="${4:?}"
+  output_dir="${5:?}"
+
+  ext="${ext:+.$ext}"
+  target="${what^^}_NAME"
+
+  local versioned_symlink_prefix="${symlink_prefix}-${version_id}"
+  local friendly_versioned_symlink_prefix="${symlink_prefix}-v${version_id}"
+
+  local os_image_symlink="${symlink_prefix}"
+  local os_image_versioned_symlink="${versioned_symlink_prefix}"
+  local os_image_friendly_versioned_symlink="${friendly_versioned_symlink_prefix}"
```

Comment:
All of these specific variables can be factored into a suffix that gets appended to the prefix:
```
local suffix
case "${what}" in
  os_image) suffix="" ;;
  data_image) suffix="-data" ;;
  boot_image) suffix="-boot.ext4.lz4" ;;
  verity_image) suffix="-root.verity.lz4" ;;
  root_image) suffix="-root.ext4.lz4" ;;
  *) echo "unknown image "${what}" >&2 ; exit 1 ;;
esac

image_symlink="${symlink_prefix}${suffix}"
image_versioned_symlink="${versioned_symlink_prefix}${suffix}"
...
```